### PR TITLE
Unit tests for CommandLineBuilders

### PR DESCRIPTION
**Pull request by sharathmalladi**

### CODE REVIEW

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/package-info.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/**
+ * Unit tests for REEF Driver running under Azure Batch runtime.
+ */
+package org.apache.reef.runtime.azbatch.driver;
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/package-info.java]: This package name and package path seems not match.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package org.apache.reef.runime.azbatch;
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java:1]: This file is using 4 indents format. While most other files in reef are using 2 indents format.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        JobSubmissionEvent event = mock(JobSubmissionEvent.class);
+
+        Optional<Integer> memory = Optional.of(100);
+        when(event.getDriverMemory()).thenReturn(memory);
+
+        String actual = this.windowsCommandBuilder.build(event);
+        String expected = "powershell.exe /c \"Add-Type -AssemblyName System.IO.Compression.FileSystem;  [System.IO.Compression.ZipFile]::ExtractToDirectory(\\\"$env:AZ_BATCH_TASK_WORKING_DIR\\local.jar\\\", \\\"$env:AZ_BATCH_TASK_WORKING_DIR\\reef\\\");  java -Xmx100m -XX:PermSize=128m -XX:MaxPermSize=128m -ea -classpath 'c:\\driverpath1;c:\\driverpath2;reef/local/*;reef/global/*;driverclasspathsuffix' -Dproc_reef org.apache.reef.runtime.common.REEFLauncher reef/local/driver.conf 1> stdout.txt 2> stderr.txt\";";
+        Assert.assertEquals(expected, actual);
+    }
+}
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java]: New line at end of file.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Test
+    public void LinuxCommandBuilderTest() throws InjectionException {
+        JobSubmissionEvent event = mock(JobSubmissionEvent.class);
+
+        Optional<Integer> memory = Optional.of(100);
+        when(event.getDriverMemory()).thenReturn(memory);
+
+        String actual = this.linuxCommandBuilder.build(event);
```
Comment (homezcx) [inline lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java]: FYI: I merged a change which renames the build interface. As long as it merges well, I think it is OK. I also add an interface to build evaluator command. I will add that interface test after you merged.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java (hunk 2 above)
Comment (sharathmalladi) [reply lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java:1]: Fixed it here and in my Intellij settings.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java (hunk 3 above)
Comment (sharathmalladi) [reply lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java]: Fixed it here and in my Intellij settings.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/package-info.java (hunk 1 above)
Comment (sharathmalladi) [reply lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/package-info.java]: Fixed.

File: lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  @Test
+  public void WindowsCommandBuilderTest() throws InjectionException {
+    JobSubmissionEvent event = mock(JobSubmissionEvent.class);
+
+    Optional<Integer> memory = Optional.of(100);
+    when(event.getDriverMemory()).thenReturn(memory);
+
+    String actual = this.windowsCommandBuilder.build(event);
```
Comment (tyclintw) [inline lang/java/reef-runtime-azbatch/src/test/java/org/apache/reef/runime/azbatch/CommandBuilderTests.java:63]: FYI - You need to update and create buildDriverCommand and buildEvaluatorCommand.

**Review by tyclintw — APPROVED** [review]

**Pull request closed without merging**